### PR TITLE
docs: update FAQ docs

### LINKED ISSUE #6594: docs: Restructure the FAQ section
### Improved or typo fixed.

The current "[FAQ](https://apisix.apache.org/docs/apisix/next/FAQ)" page misses internal hyperlinks and could use some rewriting to enhance readability.

### Solution

Make changes to the doc similar to this PR: https://github.com/apache/apisix-website/pull/893

Please assign this issue to me. I will open a PR to discuss the changes.

### PR DESCRIPTION
**Pull request by navendu-pottekkat**
Signed-off-by: Navendu Pottekkat <navendupottekkat@gmail.com>

### Description

This PR updates the FAQ section to be more readable and consistent. 

Fixes #6594

### CODE REVIEW

**Comment by navendu-pottekkat** [comment]
@juzhiyuan @yzeng25 @avinal Could you review this PR?

File: docs/en/latest/FAQ.md
Comment (spacewander) [inline docs/en/latest/FAQ.md:471]:
```suggestion
**Note**: By using the default token, you could be exposed to security risks. It is required to update it when deploying to a production environment.
```
would that be better?

File: docs/en/latest/FAQ.md
Comment (yzeng25) [inline docs/en/latest/FAQ.md:40]: Could you double check on the QPS and the latency stats? I remember the stats are 18,000 QPS and 0.2 ms latency on the [GitHub intro page](https://github.com/apache/apisix).

image

File: docs/en/latest/FAQ.md
Comment (yzeng25) [inline docs/en/latest/FAQ.md:48]:
```suggestion
## Can I write my own Plugins for Apache APISIX?
```

File: docs/en/latest/FAQ.md
Comment (yzeng25) [reply docs/en/latest/FAQ.md:40]: I mean either of them or all of them could be right. But we need to keep stats correct.

File: docs/en/latest/FAQ.md
Comment (navendu-pottekkat) [reply docs/en/latest/FAQ.md:35]: @spacewander I did not change what was already existing there.

What do you suggest we add here instead?

File: docs/en/latest/FAQ.md
Comment (navendu-pottekkat) [reply docs/en/latest/FAQ.md:28]: @nic-chen What do you suggest we add instead?

File: docs/en/latest/FAQ.md
Comment (bisakhmondal) [reply docs/en/latest/FAQ.md:35]: `This eliminates the need for polling` - Let's rephrase this. Polling is one of many options in RDBs but they do support event-based push. For eg., Postgres has 1. logical replication and 2. PG_NOTIFY to support real-time event notification. But they are pretty hard to scale compared to etcd - that might be the issue (at least that's what I think).

File: docs/en/latest/FAQ.md
Comment (avinal) [reply docs/en/latest/FAQ.md:28]: Maybe pick theses headers: https://github.com/apache/apisix#features

Comment (nic-chen) [reply docs/en/latest/FAQ.md:28]: > Maybe pick theses headers: [apache/apisix#features](https://github.com/apache/apisix#features)

yes, the `Full Dynamic` section

**Comment by navendu-pottekkat** [comment]
@spacewander @nic-chen @yzeng25 @bisakhmondal @avinal I have made the changes as suggestions.

**Review by yzeng25 — APPROVED** [review]
LGTM

**Review by nic-chen — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]
LGTM, after this PR got merged, @guoqqqi please process the Chinese version 🙏

**Comment by guoqqqi** [comment]
I will create issues, and find a new contributor to complete it 😄

**Pull request merged**
